src/task/collection.coffee

```diff
@@ -67,6 +81,7 @@ getModuleInfo = (taskId, cnxUrl = '') ->
   moduleInfo
 
 api.channel.on("task.*.receive.*", update)
+api.channel.on('task.*.receive.failure', checkFailure)
 
```

Comment:
`update` will match all `*.receive.*` events, that means it'll also get the `failure` ones.   I don't think it can handle a 500 error since it blindly accesses `eventData` (which would be empty for a "real" error).